### PR TITLE
Oskar backend

### PR DESCRIPTION
**Pull request by xDekann**
Endpoints for user deletion/edit with tests, explanation and a few fixes

### CODE REVIEW

File: gateway/src/main/resources/application.properties [hunk 1]
```diff
@@ -9,4 +9,4 @@ spring.cloud.gateway.routes[0].predicates[0]=Path=/client/**
 spring.cloud.gateway.routes[1].uri=lb://editorial
 spring.cloud.gateway.routes[1].predicates[0]=Path=/editorial/**
 
-#spring.profiles.active=dev
\ No newline at end of file
+spring.profiles.active=dev
```
Comment (SzymekN) [inline gateway/src/main/resources/application.properties:12]:
```suggestion
#spring.profiles.active=dev
```

**Review by SzymekN — APPROVED** [review]

File: client/src/main/java/com/client/controller/UserActionController.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            try {
+                User userToEdit = userRepository.findUserById(userId);
+                userActionService.updateUser(userToEdit, userRegistrationDto);
+                return ResponseEntity.ok().build();
+            } catch (Exception exception) {
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+            }
+        }
+    }
+}
```
Comment (RafMat159) [inline client/src/main/java/com/client/controller/UserActionController.java:99]: Shouldn't we add functions to UserActionService that call methods from UserRepository such as deleteUserById?

File: client/src/main/java/com/client/repository/UserRepository.java [hunk 3]
```diff
@@ -18,4 +20,12 @@ public interface UserRepository extends JpaRepository<User, Long> {
             "WHERE ud.email = :email")
     boolean existsUsersByEmail(@Param("email") String email);
 
+    @Modifying
+    @Transactional
+    @Query("DELETE FROM User u WHERE u.id =:id")
+    void deleteUserById(Long id);
```
Comment (RafMat159) [inline client/src/main/java/com/client/repository/UserRepository.java:26]: If you add UserRepository to the UserActionService, we will move this annotation(@Transactional) only to this layer.
https://vladmihalcea.com/spring-transactional-annotation/

File: client/src/main/java/com/client/service/UserActionServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                .queryParam("id", userId).build().toUri();
+        return restTemplate.exchange(endpointUri, HttpMethod.DELETE, new HttpEntity<>(null, headers), String.class);
+    }
+
+    @Override
+    public ResponseEntity<String> updateUser(User user, UserRegistrationDto userRegistrationDto) {
+        editUserByDto(user, userRegistrationDto);
+        userRepository.save(user);
+        return ResponseEntity.ok("Successfully saved user");
+    }
```
Comment (RafMat159) [inline client/src/main/java/com/client/service/UserActionServiceImpl.java:59]: Isn't it better to return the void type?

File: client/src/test/java/com/client/service/UserActionServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    private UserRegistrationDto userRegistrationDto;
+    private HttpServletRequest request;
+    private User user;
+    @BeforeAll
+    void setUp() {
+        userRegistrationDto = mock(UserRegistrationDto.class);
+        userActionService = new UserActionServiceImpl(userRepository, basicService, passwordEncoder);
+        request = mock(HttpServletRequest.class);
+        user = mock(User.class);
+    }
```
Comment (RafMat159) [inline client/src/test/java/com/client/service/UserActionServiceImplTest.java:44]: You can simplify this code by using annotations such as:
@ExtendWith(MockitoExtension.class), @Mock and  @InjectMocks

File: editorial/src/main/java/com/editorial/controller/UserActionController.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            return ResponseEntity.badRequest().build();
+        else {
+            try {
+                User userToEdit = userRepository.findUserById(userId);
+                userActionService.updateUser(userToEdit, userRegistrationDto);
+                return ResponseEntity.ok().build();
+            } catch (Exception exception) {
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+            }
+        }
```
Comment (RafMat159) [inline editorial/src/main/java/com/editorial/controller/UserActionController.java:97]: Same situation as in microservice client.

File: editorial/src/main/java/com/editorial/repository/UserRepository.java [hunk 7]
```diff
@@ -14,4 +16,13 @@ public interface UserRepository extends JpaRepository<User, Long> {
     @Query("SELECT CASE WHEN COUNT(u) > 0 THEN true ELSE false END FROM User u JOIN u.userDetails ud " +
             "WHERE ud.email = :email")
     boolean existsUsersByEmail(@Param("email") String email);
+
+    @Modifying
+    @Transactional
+    @Query("DELETE FROM User u WHERE u.id =:id")
+    void deleteUserById(Long id);
+
```
Comment (RafMat159) [inline editorial/src/main/java/com/editorial/repository/UserRepository.java:24]: Same situation as in microservice client
https://vladmihalcea.com/spring-transactional-annotation/

File: editorial/src/main/java/com/editorial/service/UserActionServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+                .queryParam("id", userId).build().toUri();
+        return restTemplate.exchange(endpointUri, HttpMethod.DELETE, new HttpEntity<>(null, headers), String.class);
+    }
+
+    @Override
+    public ResponseEntity<String> updateUser(User user, UserRegistrationDto userRegistrationDto) {
+        editUserByDto(user, userRegistrationDto);
+        userRepository.save(user);
+        return ResponseEntity.ok("Successfully saved user");
+    }
```
Comment (RafMat159) [inline editorial/src/main/java/com/editorial/service/UserActionServiceImpl.java:58]: Isn't it better to return the void type?

**Review by RafMat159 — APPROVED** [review]
Looks good

**Pull request merged**